lib/index.js

```diff
@@ -313,32 +306,29 @@ class Mcdev {
      */
     static async document(businessUnit, type) {
         Util.logger.info('mcdev:: Document');
+
+        const properties = await config.getProperties();
+        if (type && !MetadataTypeInfo[type]) {
+            Util.logger.error(`:: '${type}' is not a valid metadata type`);
+            return;
+        }
         try {
-            const properties = await config.getProperties();
-            if (type && !MetadataTypeInfo[type]) {
-                Util.logger.error(`:: '${type}' is not a valid metadata type`);
-                return;
-            }
-            try {
-                const parentBUOnlyTypes = ['accountUser', 'role'];
-                const buObject = await Cli.getCredentialObject(
-                    properties,
-                    parentBUOnlyTypes.includes(type) ? businessUnit.split('/')[0] : businessUnit,
-                    parentBUOnlyTypes.includes(type) ? Util.parentBuName : null
-                );
-                if (buObject !== null) {
-                    MetadataTypeInfo[type].properties = properties;
-                    MetadataTypeInfo[type].document(buObject);
-                }
-            } catch (ex) {
-                Util.logger.error('mcdev.document ' + ex.message);
-                Util.logger.debug(ex.stack);
-                Util.logger.info(
-                    'If the directoy does not exist, you may need to retrieve this BU first.'
-                );
+            const parentBUOnlyTypes = ['accountUser', 'role'];
+            const buObject = await Cli.getCredentialObject(
+                properties,
+                parentBUOnlyTypes.includes(type) ? businessUnit.split('/')[0] : businessUnit,
+                parentBUOnlyTypes.includes(type) ? Util.parentBuName : null
+            );
+            if (buObject !== null) {
+                MetadataTypeInfo[type].properties = properties;
+                MetadataTypeInfo[type].document(buObject);
             }
         } catch (ex) {
-            Util.logger.error(`Document failed: ${ex.message}`);
+            Util.logger.error('mcdev.document ' + ex.message);
+            Util.logger.debug(ex.stack);
```

Comment:
please use this instead
https://github.com/Accenture/sfmc-devtools/blob/bc01163fe4a30c16a00f660497c585aaed13bf96/lib/util/util.js#L474